log_forwarded_for_ip/models/log_forwarded_for_ip_installed.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# © 2015 Aserti Global Solutions
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from werkzeug.serving import WSGIRequestHandler
+from mimetools import Message
+from openerp import models
+
+
+def address_string(self):
+    if self.headers and isinstance(self.headers, Message):
+        forwarded_for = self.headers.get('X-Forwarded-For', '').split(',')
+        if forwarded_for and forwarded_for[0]:
+            return forwarded_for[0]
+    return self.client_address[0]
+
+
+class log_forwarded_for_ip_installed(models.AbstractModel):
```

Comment:
images key is only a warning, not mandatory, but it makes sense because new Odoo apps put the first image as background of the tile, so if we want to make branding, we need to make good modules with some "marketing" material. But as stated, this is only a recommendation, not a requirement.